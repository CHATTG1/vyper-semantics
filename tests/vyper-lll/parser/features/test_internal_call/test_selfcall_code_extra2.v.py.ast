%pgm(, ,
  %fdecl(%@public, foo, %param(x, %bytesT(5)) %param(y, %bytesT(5)), %void,
    %pass)
  %fdecl(%@public, bar, , %void,
    %stmtexpr(%icall(foo, "abc" "abc")))
)
