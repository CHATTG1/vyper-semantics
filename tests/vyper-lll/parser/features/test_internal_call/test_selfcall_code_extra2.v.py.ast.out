$%seq ( $return ( 0 , $lll ( $%seq ( $%seq ( $mstore ( 28 , $calldataload ( 0 ) ) , $mstore ( 32 , 1461501637330902918203684832716283019655932542976 ) , $mstore ( 64 , 170141183460469231731687303715884105727 ) , $mstore ( 96 , -170141183460469231731687303715884105728 ) , $mstore ( 128 , 1701411834604692317316873037158841057270000000000 ) , $mstore ( 160 , -1701411834604692317316873037158841057280000000000 ) ) , $if ( $eq ( $mload ( 0 ) , 1703115483 ) , $%seq ( $calldatacopy ( 320 , 4 , 64 ) , $assert ( $iszero ( $callvalue ) ) , $%seq ( $calldatacopy ( 384 , $add ( 4 , $calldataload ( $add ( 4 , 0 ) ) ) , 37 ) , $assert ( $le ( $calldataload ( $add ( 4 , $calldataload ( $add ( 4 , 0 ) ) ) ) , 5 ) ) ) , $%seq ( $calldatacopy ( 480 , $add ( 4 , $calldataload ( $add ( 4 , 32 ) ) ) , 37 ) , $assert ( $le ( $calldataload ( $add ( 4 , $calldataload ( $add ( 4 , 32 ) ) ) ) , 5 ) ) ) , $pass , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 4273672062 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $assert ( $call ( $gas , $address , 0 , $with ( $_poz , 64 , $%seq ( $mstore ( 512 , 1703115483 ) , $mstore ( 544 , $_poz ) , $with ( $_s , $%seq ( $mstore ( 320 , 3 ) , $mstore ( $add ( 320 , 32 ) , 44048180597813453602326562734351324025098966208897425494240603688123167145984 ) , 320 ) , $%seq ( $with ( $_source , $_s , $with ( $_sz , $add ( 32 , $mload ( $_source ) ) , $assert ( $call ( $add ( 18 , $div ( $_sz , 10 ) ) , 4 , 0 , $_source , $_sz , $add ( 544 , $_poz ) , $_sz ) ) ) ) , $set ( $_poz , $add ( 32 , $add ( $_poz , $mload ( $_s ) ) ) ) ) ) , $mstore ( 576 , $_poz ) , $with ( $_s , $%seq ( $mstore ( 416 , 3 ) , $mstore ( $add ( 416 , 32 ) , 44048180597813453602326562734351324025098966208897425494240603688123167145984 ) , 416 ) , $%seq ( $with ( $_source , $_s , $with ( $_sz , $add ( 32 , $mload ( $_source ) ) , $assert ( $call ( $add ( 18 , $div ( $_sz , 10 ) ) , 4 , 0 , $_source , $_sz , $add ( 544 , $_poz ) , $_sz ) ) ) ) , $set ( $_poz , $add ( 32 , $add ( $_poz , $mload ( $_s ) ) ) ) ) ) , 540 ) ) , 196 , 0 , 0 ) ) , $stop ) ) ) , 0 ) ) )
